lib/statesman/adapters/active_record.rb

```diff
@@ -65,18 +65,18 @@ def history(force_reload: false)
         end
       end
 
-      # rubocop:disable Naming/MemoizedInstanceVariableName
       def last(force_reload: false)
         if force_reload
           @last_transition = history(force_reload: true).last
+        elsif instance_variable_defined?(:@last_transition)
+          @last_transition
         else
-          @last_transition ||= history.last
```

Comment:
tl;dr; this caching approach doesn't work when `history.last` returns `nil`

---

When the object is just created and there are no sate transitions, `history.last` returns `nil` because:

```
transitions_for_parent.order(:sort_key) => []
transitions_for_parent.order(:sort_key).last => nil
```

and then it's stored in `@last_transition` variable. 

If we call `#last` again (which happens when there are multiple `guard_transition` against the initial state transition) this statement:

```
@last_transition ||= history.last
```

will call `history.last` again which would result in the database query (which will return zero rows).